src/components/NewCommentForm/NewCommentForm.jsx

```diff
@@ -1,39 +1,90 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { addComment } from '../../api/comments';
 import './NewCommentForm.scss';
 
-export const NewCommentForm = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+export const NewCommentForm = ({ selectedPostId, onRefreshDetails }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const resetForm = () => {
+    setName('');
+    setEmail('');
+    setBody('');
+  };
+
+  const setValue = (event, setter) => {
+    setter(event.target.value);
+  };
+
+  const submitComment = (event) => {
+    event.preventDefault();
+    addComment({
+      name,
+      email,
+      body,
+      postId: selectedPostId,
+    })
+      .then(() => onRefreshDetails());
+    resetForm();
+  };
+
+  return (
+    <form
+      className="NewCommentForm"
+      onSubmit={submitComment}
     >
-      Add a comment
-    </button>
-  </form>
-);
+      <div className="form-field">
+        <input
+          type="text"
+          name="name"
+          placeholder="Your name"
+          className="NewCommentForm__input"
+          value={name}
+          onChange={(event) => {
+            setValue(event, setName);
+          }}
```

Comment:
I think it'd be the same if you just write setName for this field, but now you have +1 function